game/src/game/components/Controllers/AIController.cpp

```diff
@@ -0,0 +1,85 @@
+#include "AIController.h"
+
+#include "engine/core/debug/Log.h"
+#include "engine/physics/PhysicsService.h"
+#include "engine/scene/Entity.h"
+
+AIController::AIController()
+    : input_service_(nullptr),
+      transform_(nullptr),
+      ai_service_(nullptr)
+{
+}
+
+void AIController::OnInit(const ServiceProvider& service_provider)
+{
+    input_service_ = &service_provider.GetService<InputService>();
+    ai_service_ = &service_provider.GetService<AIService>();
+    transform_ = &GetEntity().GetComponent<Transform>();
+    GetEventBus().Subscribe<OnUpdateEvent>(this);
+
+    if (ai_service_)
+    {
+        // store the path in a local variable.
+        path_to_follow_ = ai_service_->GetPath();
+    }
```

Comment:
No need to check for `if (ai_service_)`. If the service provider can't find it, it'll assert and crash the game